pkg/agent/controller/bgp/controller.go

```diff
@@ -991,3 +983,94 @@ func (c *Controller) GetBGPPeerStatus(ctx context.Context) ([]bgp.PeerStatus, er
 	}
 	return peers, nil
 }
+
+// GetBGPRoutes returns the advertised/received routes to/from the given peerAddress.
+// If given peerAddress is empty, routes from all BGP Peers are returned.
+func (c *Controller) GetBGPRoutes(ctx context.Context, routeType bgp.RouteType, peerAddress string) (sets.Set[string], error) {
+	bgpPeerAddresses := c.getBGPPeerAddresses()
+	if bgpPeerAddresses == nil {
+		return nil, ErrBGPPolicyNotFound
+	}
+
+	// Validate peerAddress, consider empty peerAddress as valid.
+	if peerAddress != "" {
+		if !utilnet.IsIPv6String(peerAddress) && !utilnet.IsIPv4String(peerAddress) {
+			return nil, fmt.Errorf("invalid peer address: %s", peerAddress)
+		}
+		if !slices.Contains(bgpPeerAddresses, peerAddress) {
+			return nil, fmt.Errorf("bgp peer %s not found", peerAddress)
+		}
+		bgpPeerAddresses = []string{peerAddress}
+	}
+
+	bgpRoutes := make(sets.Set[string])
+	if routeType == bgp.RouteAdvertised {
+		// Advertised routes are same for all bgp peers.
+		bgpRoutesAdvertised := c.getBgpRoutesAdvertised()
+		if bgpRoutesAdvertised == nil {
+			return nil, ErrBGPPolicyNotFound
+		}
+		for route := range bgpRoutesAdvertised {
+			bgpRoutes.Insert(route.Prefix)
+		}
+	} else {
+		bgpRoutesReceived, err := c.getBgpRoutesReceived(ctx, bgpPeerAddresses)
+		if err != nil {
+			return nil, err
+		}
+		for route := range bgpRoutesReceived {
+			bgpRoutes.Insert(route.Prefix)
+		}
+	}
```

Comment:
Since we don't need to handle received routes, you can delete from L1016 to L1024.